### ISSUE TITLE
mc mirror with ceph rgw as a source fails with --watch option (SignatureDoesNotMatch)

### ISSUE THREAD

**Issue by lgyurci** [minio/mc#4916]
## Expected behavior
Well, technically mc does not support anything other s3 compatible provider than minio or aws, but I've seen issues getting resolved regarding to ceph-mc compatibility, so I decided I'll give this a shot. mc mirror does work work with ceph rgw without the --watch option (so subscribing to bucket notifications is the problem), but I believe technically it should with the --watch too.
## Actual behavior
Ceph rgw throws back a 403 forbidden error, along with "SignatureDoesNotMatch". mc output (json):
```
{
 "status": "error",
 "error": {
  "message": "Failed to perform mirroring",
  "cause": {
   "message": "The request signature we calculated does not match the signature you provided. Check your key and signing method.",
   "error": {
    "Code": "SignatureDoesNotMatch",
    "Message": "",
    "BucketName": "",
    "Key": "",
    "Resource": "",
    "RequestID": "tx00000bfc33392d1d3f361-00663362b8-8a54f9-ceph-store",
    "HostID": "8a54f9-ceph-store-ceph-store",
    "Region": "",
    "Server": ""
   }
  },
  "type": "error",
  "trace": [
   {
    "line": 931,
    "file": "client-s3.go",
    "func": "cmd.(*S3Client).Watch.func1"
   },
   {
    "line": 574,
    "file": "mirror-main.go",
    "func": "cmd.(*mirrorJob).monitorMirrorStatus"
   }
  ],
  "sysinfo": {
   "host.arch": "amd64",
   "host.cpus": "24",
   "host.lang": "go1.21.9",
   "host.name": "redacted",
   "host.os": "linux",
   "mem.heap.total": "19 MiB",
   "mem.heap.used": "9.9 MiB",
   "mem.total": "28 MiB",
   "mem.used": "9.9 MiB"
  }
 }
}

```
## Steps to reproduce the behavior
Use mc mirror on a ceph rgw as a source with --watch
## mc --version
mc version RELEASE.2024-04-29T09-56-05Z (commit-id=5b7b2223717a32ff01d63d57c2d040a719ca581e)
Runtime: go1.21.9 linux/amd64
## System information
The whole setup is running on a kubernetes cluster (ceph, and the mc mirror client as well). The ceph rgw pod ip and port is statically baked into the mc config.json, and everything else seems to work.

**Comment by harshavardhana** [comment]
--watch is MinIO specific

**Comment by harshavardhana** [comment]
> Failed to perform mirroring Listening for bucket notification is specific only to `minio` server endpoints

We return this error when the endpoints are either S3 or GCS (S3 endpoint)

**Comment by lgyurci** [comment]
Where exactly should this error be displayed? Because I did not see it in any logs. I might have missed it though. 

Also, imo this should be mentioned in the docs (https://min.io/docs/minio/linux/reference/minio-mc/mc-mirror.html)

And I also found ceph specific examples on the minio blog, where they used the --watch option (with ceph as the source, I guess): https://blog.min.io/data-migration-tools-into-minio/ :

> You can then use mc mirror to copy the data to your MinIO cluster
`mc mirror ceph/mydata destminio/mydata`
We suggest that you run the mc mirror command with the --watch flag to continuously monitor for objects and sync them to MinIO.
